src/pybind/mgr/mds_autoscaler/module.py

```diff
@@ -32,12 +32,12 @@ def get_service(self, fs_name: str) -> Optional[orchestrator.ServiceDescription]
             return completion.result[0]
         return None
 
-    def update_daemon_count(self, spec: ServiceSpec, fs_name: str, abscount: int) -> ServiceSpec:
+    def update_daemon_count(self, spec: MDSSpec, fs_name: str, abscount: int) -> MDSSpec:
```

Comment:
Still getting mypy failures causing make check to fail. I think we need to cast from service spec to mds spec here https://github.com/ceph/ceph/blob/master/src/pybind/mgr/mds_autoscaler/module.py#L82

Reply:
I fixed the signature of update_daemon_count so now it uses ServiceSpec instead of MDSSpec:

`def update_daemon_count(self, spec: ServiceSpec, fs_name: str, abscount: int) -> MDSSpec:`

this fixes the mypy error